webapp/src/ts/components/header/header.component.ts

```diff
@@ -75,19 +75,25 @@ export class HeaderComponent implements OnInit, OnDestroy {
   }
 
   openGuidedSetup() {
-    this.modalService.show(GuidedSetupComponent);
+    this.modalService
+      .show(GuidedSetupComponent)
+      .catch(() => {});
   }
 
   openTourSelect() {
 
   }
 
   openFeedback() {
-    this.modalService.show(FeedbackComponent);
+    this.modalService
+      .show(FeedbackComponent)
+      .catch(() => {});
```

Comment:
there are no errors here. Please read this comment carefully: https://github.com/medic/cht-core/blob/4750-navigation-confirm-modal/webapp/src/ts/modals/mm-modal/mm-modal.ts#L47

Reply:
the reason why the `catch` block exists is so our ExceptionHandler doesn't report modal rejections are unhandled rejections and create feedback documents. 

I'm going to try to see what happens if the modal code breaks, if this comes out as an exception that's caught by this block or not.

Reply:
So, I made the `modalService` throw an error before returning the promise:
![image](https://user-images.githubusercontent.com/35681649/99280964-a5372180-283a-11eb-8334-58a7f3b22585.png)

I made the `FeedbackComponent` constructor to throw an error:
![image](https://user-images.githubusercontent.com/35681649/99281415-22629680-283b-11eb-81c3-3d82e9e1ae49.png)

I added a `ngOnInit` function to `FeedbackComponent` to just make it throw an error:
![image](https://user-images.githubusercontent.com/35681649/99281736-8ab17800-283b-11eb-826a-b682c3f8213d.png)

I made the `submit` function of `FeedbackComponent` throw an error: 
![image](https://user-images.githubusercontent.com/35681649/99282084-ec71e200-283b-11eb-8148-d35a8b87d755.png)

I'm confident enough now that no uncaught errors will be swallowed by that catch :)

Reply:
> there are no errors here. Please read this comment carefully: https://github.com/medic/cht-core/blob/4750-navigation-confirm-modal/webapp/src/ts/modals/mm-modal/mm-modal.ts#L47

The link got a 404 error I think because the branch was deleted. Yes maybe it can be resolved in another PR later, but I don't think is a good idea to silent errors, moreover if the errors comes from another component that the modal service only fire, but don't control (but I not sure what was the argument in that comment though).

Reply:
I took a look to the comment in the branch were it was merged :smile: :

```
    // attach a promise to the BsModalRef (we can access this promise from BsModalRef.content.modalClosePromise)
    // this promise is resolved when the user "submits" or accepts the modal
    // this promise is rejected when the user "cancels" or dismisses the modal (any action that hides the modal that is
    // not clicking the submit button)
```

I understand that the original promise is accessible to evaluate if needed, but that does not avoid that in some parts the exceptions are muted and there is no part now where they are evaluated using that reference, right? Like we can, but we are not doing it now.

What I think, and maybe we can discuss it in a new issue under the angular migration project, is that cancel a window shouldn't fire a exception, instead it should resolve the promise with a value like `{ exitAction: 'cancel' }` when user did cancel the modal, or `{ exitAction: 'accept' }` when the user pressed the "Accept" button. So rejected promise should be used only when there was an exception in the code that was not catched. What you think?

Reply:
>  maybe we can discuss it in a new issue under the angular migration project
Feel free to add the new issue to the project!